### ISSUE TITLE
Continuous Identification for burger equation

### ISSUE THREAD

**Issue by abbasi-el** [janblechschmidt/PDEsByNNs#1]
I am using your code for finding unknown parameters lambda_1 and lambda_2 of burger equation in continuous times. But the results for lambda_1 and lambda_2 is strange. I would appreciate if you could provide continuous identification for the Burger unknown parameters or let me know the problem of the code.

class PINNIdentificationNet(PINN_NeuralNet):
    def __init__(self, *args, **kwargs):
        
        # Call init of base class
        super().__init__(*args,**kwargs)

        # Initialize parameters
        self.lambda_1 = tf.Variable([0.0], trainable=True,dtype=DTYPE)
        self.lambda_2 = tf.Variable([-6.0], trainable=True,dtype=DTYPE)
        self.lambd_1list = []
        self.lambd_2list = []
    def get_params(self, numpy=False):
          l1 = self.lambda_1
          l2 = tf.exp(self.lambda_2)
          if numpy:
            return l1.numpy()[0], l2.numpy()[0]
          return l1, l2 

class BurgerPINNIdentification(PINNSolver):
    
    def fun_r(self, t, x, u, u_t, u_x, u_xx):
        """Residual of the PDE"""
        l1, l2 = self.model.get_params()
        return u_t + l1 * u * u_x - tf.exp(l2) * u_xx
        
    def callback(self, xr=None):
        lambda_1 = self.model.lambda_1.numpy()[0]
        lambda_2= self.model.lambda_2.numpy()[0]
        self.model.lambd_1list.append(lambda_1)
        self.model.lambd_2list.append(lambda_2)
        
        if self.iter % 50 == 0:
            print('It {:05d}: loss = {:10.8e} l1: {:10.5e} l2: {:10.5e}'.format(self.iter, self.current_loss, lambda_1,np.exp(lambda_2)))
            

        
        self.hist.append(self.current_loss)
        self.iter += 1
        
    def plot_loss_and_param(self, axs=None):
        if axs:
            ax1, ax2 = axs
            self.plot_loss_history(ax1)
        else:
            ax1 = self.plot_loss_history()
            ax2 = ax1.twinx()  # instantiate a second axes that shares the same x-axis

        color = 'tab:blue'
        ax2.tick_params(axis='y', labelcolor=color)
        ax2.plot(range(len(self.hist)), self.model.lambd_1list,'-',color=color)
        ax2.set_ylabel('$\\lambda^{n_{epoch}}$', color=color)
        return (ax1,ax2)

from time import time
# Initialize model
model = PINNIdentificationNet(lb, ub, num_hidden_layers=8,
                                        num_neurons_per_layer=20,
                                        activation='tanh',
                                        kernel_initializer='glorot_normal')

model.build(input_shape=(None,2))

# Initilize solver
burgerPINNIdentification = BurgerPINNIdentification(model, X_r)

# Choose step sizes aka learning rate
#lr = tf.keras.optimizers.schedules.PiecewiseConstantDecay([3000,7000],[1e-1,1e-2,1e-3])
lr = tf.keras.optimizers.schedules.PiecewiseConstantDecay([1000,3000],[1e-2,1e-3,5e-4])
# Solve with Adam optimizer
optim = tf.keras.optimizers.Adam(learning_rate=lr)

# Start timer
t0 = time()
# burgerPINNIdentification.solve_with_TFoptimizer(optim, X_data, u_data, N=8000) 
burgerPINNIdentification.solve_with_ScipyOptimizer(X_data, u_data,
                            method='L-BFGS-B',
                            options={'maxiter': 50000,
                                     'maxfun': 50000,
                                     'maxcor': 50,
                                     'maxls': 50,
                                     'ftol': 1.0*np.finfo(float).eps})
# Print computation time
print('\nComputation time: {} seconds'.format(time()-t0))

**Comment by janblechschmidt** [comment]
Dear @abbasi-el,
unfortunately, I can't reproduce your results. First of all, you have to make sure that you provide some measurements from your solution (computed with the parameters that you want to identify). Since you do not specify, where this information comes from, I can't help you at the moment. In contrast to the eikonal equation example, there is no closed form solution for  the Burgers equation.
Best regards,
Jan
